website/blog/2018-05-29-jest-23-blazing-fast-delightful-testing.md

```diff
@@ -0,0 +1,124 @@
+
+![jest-each in action](/jest/img/blog/23-jest-each.png)
+
+Huge shout out to Prettier for [fixing](https://github.com/prettier/prettier/pull/4423) the template string formatting so quickly ([see 1.13 Prettier post](https://prettier.io/blog/2018/05/23/1.13.0.html#format-new-describeeach-table-in-jest-23-4423-by-ikatyang))!
+
+## New Matchers
+
+We only add matchers to core if we believe they will be useful to a large amount of people in the Jest community, and leave the majority of matchers to the community (see [jest-extended](https://yarnpkg.com/en/package/jest-extended)). Some matchers make the cut into core, and Jest 23 adds:
+
+* nthCalledWith
+* toReturn
+* toReturnTimes
+* toReturnWith
+* lastReturnedWith
+* nthReturnedWith
+* toStrictEqual
+
+![New matchers in action](/jest/img/blog/23-new-matchers.png)
+
+See the updated expect docs [here](https://facebook.github.io/jest/docs/en/expect.html).
+
+## Debug Hanging Tests
+
+A common issue we see on the issue tracker relates to “Jest” hanging after a test run. This is usually due to app code leaving handles open, preventing Jest from exiting. In the past, users have resorted to `—forceExit` to fix (not recommended).
+
+To help debug these issues, we're now detecting when Jest does not exit:
+
+![Detecting hanging tests](/jest/img/blog/23-hanging-before.png)
+
+And we offer a new flag `--detectOpenHandles` to help find the open handles:
+
+![Running detectOpenHandles](/jest/img/blog/23-hanging-after.png)
+
+See the updated CLI docs [here](https://facebook.github.io/jest/docs/en/cli.html#detectopenhandles).
+
+## Watch Mode Plugins
+
+We have completely rewritten the watch mode system to allow adding custom plugins to watch mode. Watch Mode Plugins now make it possible to hook into Jest events and provide custom menu options in the Watch Mode Menu. All of the default Watch Mode prompts are implemented as plugins in this system, and the docs to create your own are available [here](https://facebook.github.io/jest/docs/en/watch-plugins.html).
+
+With this change, we're also now able to bring back typeahead support as a Watch Mode Plugin via [](https://github.com/rogeliog/jest-plugin-filename)[jest-plugin-filename](https://github.com/jest-community/jest-plugin-filename)!
+
+![Typeahead plugin in action](/jest/img/blog/23-typeahead.gif)
+
+See [jest-plugin-filename](https://github.com/jest-community/jest-plugin-filename) for documentation and install instructions. Huge thank you to [@rogeliog](https://twitter.com/rogeliog) for the new watch mode plugin system and the jest-plugin-filename plugin!
+
+## Breaking Changes
+
+As with every major release, we are making a few breaking changes to make larger changes in the future possible and to push the testing experience to a new level. Here's a list of the biggest changes you may see:
+
+* **Require test descriptions**: We're now failing tests that do not include a description.
```

Comment:
Wasn't it test callbacks? Or is it something else than https://github.com/facebook/jest/pull/5558

Reply:
we need 2 args always, so both 🙂

Reply:
I think it's both @SimenB can you confirm?

Reply:
yes, both